kpis/kpi_calculator.py

```diff
@@ -59,6 +59,97 @@ def __init__(self, testcase):
                         'BiomassPower',
                         'SolarThermalPower', 
                         'FreshWaterFlowRate']
+        
+        # Initialize KPI Calculator variables
+        self.initialize_kpi_vars(label='tdis')
+        self.initialize_kpi_vars(label='idis')
+        self.initialize_kpi_vars(label='ener')
+        self.initialize_kpi_vars(label='cost')
+        self.initialize_kpi_vars(label='emis')
+        
+    def initialize_kpi_vars(self, label='ener'):
+        '''Initialize variables required for KPI calculation
+        
+        '''
+        # Initialize index since last integration
+        setattr(self, 'i_last_{}'.format(label), 0) 
+        # Dictionary to store energy usage by element
+        setattr(self, '{}_dict'.format(label), OrderedDict())
+        # Dictionary to store energy usage by source 
+        setattr(self, '{}_dict_by_source'.format(label), OrderedDict())
+
```

Comment:
Why the use of ``setattr`` instead of ``self.x = y``?

Reply:
This is just to enable an attribute with a dynamic name that can be defined from a string using `format()`

Reply:
Ah ok I see now.  Ok sounds good, thanks.